dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/importers/ui/ImporterForm.java

```diff
@@ -126,12 +128,22 @@ public void handleFileUpload(FileUploadEvent event) throws IOException {
         UploadedFile file = Optional.ofNullable(event)
                 .map(FileUploadEvent::getFile)
                 .orElseThrow(() -> new IllegalStateException("Null event or file"));
+        long maxUploadedFileSize = importer.getMaxUploadedFileSize();
+        if (maxUploadedFileSize != 0 && file.getSize() > maxUploadedFileSize) {
```

Comment:
I am wondering if we should validate size here or directly on xhtml. When we validate here - we technically allow any size of an file, but just not allow for it to be analyzed by generate.INP. What do you think?